### DISCUSSION TITLE
"tail sampling" processor - Beta

### DISCUSSION THREAD

**Discussion by vkrishnamoorthy-sift** [open-telemetry/opentelemetry-collector-contrib#18435] [Q&A] [answered]
I want to use tail sampling in production. The stability says Beta. When is the proposed GA date?
Also can someone point me to the pros/cons of running this in prod, any deployment architecture I can follow - capacity, performance benchmarks?

Thanks for your help, and I am new to otel-collector.

**Comment by atoulme** [comment, marked as answer]
There is no date proposed for this component. The collector as a whole is not declared stable and the tail sampling processor is depending on libraries also in beta state. There are a number of resources on the best practices to deploy and scale the collector available on opentelemetry.io: https://opentelemetry.io/docs/collector/scaling/

**Comment by jpkrohling** [comment]
Note that our definition of beta states this: 

> Same as Alpha, but the configuration options are deemed stable. While there might be breaking changes between releases, component owners should try to minimize them. A component at this stage is expected to have had exposure to non-critical production workloads already during its Alpha phase, making it suitable for broader usage.

Right now, pretty much all components are at most beta (or should be), given the underlying APIs are not stable yet. People use this in production, and we have high expectations of this component. If you use and find performance issues, do report them.
